src/redux/user/userSlice.ts

```diff
@@ -2,29 +2,43 @@ import {createSlice} from '@reduxjs/toolkit'
 import {IAuthPropsState} from '../../types/storeTypes'
 
 const initialState: IAuthPropsState = {
+  firstName: '',
+  lastName: '',
   user: '',
   email: '',
-  uid: '',
+  userId: 0,
+  authUid: '',
+  salary: 0,
+  isAuth: false,
   token: '',
-  isAuth: true,
+  avatar: 'http://res.cloudinary.com/avoo/image/upload/v1652271069/v35k9wpyszjbxhyl8xn0.jpg',
 }
 
 const userSlice = createSlice({
   name: 'user',
   initialState,
   reducers: {
     setUser(state, action) {
+      state.firstName = action.payload.firstName
+      state.lastName = action.payload.lastName
       state.user = action.payload.user
       state.email = action.payload.email
-      state.uid = action.payload.uid
+      state.authUid = action.payload.authUid
+      state.userId = action.payload.userId
+      state.isAuth = true
```

Comment:
Why always true

Reply:
When you sign in or sign up, in redux it goes to true, for loading, or we can give loading if we have userId